Tests/Tests/RouterSpec.swift

```diff
@@ -279,6 +361,38 @@ class RouterSpec: QuickSpec {
                         expect(window.findTooltipView()).toEventuallyNot(beNil())
                     }
 
+                    it("will display a tooltip above the campaign view") {
+                        let rootView = UIView()
+                        UIApplication.shared.keyWindow?.addSubview(rootView)
+                        router.accessibilityCompatibleDisplay = true
+                        let campaign = TestHelpers.generateCampaign(id: "test", type: .modal)
+                        router.displayCampaign(campaign, associatedImageData: nil, confirmation: true, completion: { _ in })
+                        expect(window.findIAMView()).toEventuallyNot(beNil())
+
+                        router.displayTooltip(tooltip,
+                                              targetView: rootView,
+                                              identifier: TooltipViewIdentifierMock,
+                                              imageBlob: imageBlob,
+                                              becameVisibleHandler: { _ in },
+                                              confirmation: true,
+                                              completion: { _ in })
+                        expect(window.findTooltipView()).toEventuallyNot(beNil())
+                        window.findTooltipView()?.removeFromSuperview()
+                        window.findIAMView()?.removeFromSuperview()
+                    }
+
+                    it("will not display a tooltip campaign with no tooltip data") {
+                        let tooltip = TestHelpers.generateTooltip(id: "test", isTooltip: false)
```

Comment:
You can use `generateCampaign` instead. `isTooltip` parameter doesn't make sense with `generateTooltip` method. How can you generate a tooltip that is not a tooltip?

Reply:
Same comment [here](https://github.com/rakutentech/ios-inappmessaging/pull/275#discussion_r1178695652), I think we need to find solution for this issue.